src/models/disassemblyoutput.cpp

```diff
@@ -207,29 +207,25 @@ DisassemblyOutput::ObjectdumpOutput DisassemblyOutput::objdumpParse(const QByteA
             continue;
         }
 
-        // detect lines like:
-        //     4f616:\t<jumps>84 c0\ttest %al,%al
-        auto lineRest = QStringView(asmLine);
-
-        // <four spaces><hex addr>:\t
-        const auto addr = [&]() -> quint64 {
-            const auto prefix = QLatin1String("    ");
-            const auto suffix = QLatin1String(":\t");
-            if (!lineRest.startsWith(prefix))
-                return 0;
+        // a line looks like this:
+        // [spaces]addr:\t [branch visualisazion] [hexdump]\tdiassembly
+        // we can simplify parsing by splitting it into three parts
+
+        const auto parts = asmLine.split(QLatin1Char('\t'));
 
-            const auto addrEnd = lineRest.indexOf(suffix, prefix.size());
-            if (addrEnd == -1)
+        const auto addr = [addrString = parts[0].trimmed(), &asmLine]() -> uint64_t {
+            const auto suffix = QLatin1Char(':');
+            if (!addrString.endsWith(suffix))
                 return 0;
 
             bool ok = false;
-            auto ret = lineRest.mid(0, addrEnd).toULongLong(&ok, 16);
+            auto ret = addrString.midRef(0, addrString.size() - 1).toULongLong(&ok, 16);
```

Comment:
the midRef isn't needed anymore now, it should be the same as just `addrString.toULongLong(...)`